pkg/cmd/infra/router/template.go

```diff
@@ -2,87 +2,163 @@ package router
 
 import (
 	"errors"
+	"fmt"
 	"strconv"
 
+	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	ktypes "k8s.io/kubernetes/pkg/types"
+
 	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/util/proc"
+	"github.com/openshift/origin/pkg/version"
 	templateplugin "github.com/openshift/origin/plugins/router/template"
+)
 
-	ktypes "k8s.io/kubernetes/pkg/types"
+const (
+	routerLong = `
+Start a router
+
+This command launches a router connected to your cluster master. The router listens for routes and endpoints
+created by users and keeps a local router configuration up to date with those changes.
+
+You may customize the router by providing your own --template and --reload scripts.
+
+You may restrict the set of routes exposed by using the --labels, --fields, or --namespace arguments.`
 )
 
-// templateRouterConfig is the config necessary to start a template router plugin.
-type templateRouterConfig struct {
-	// TemplateFile is the path to the file containing the templates for any files
-	// the template plugin should generate.
-	TemplateFile string
+type TemplateRouterOptions struct {
+	Config *clientcmd.Config
 
-	// ReloadScript is the path to the command that should be executed to reload
-	// the configuration when the plugin updates it.
-	ReloadScript string
+	TemplateRouter
+	RouterStats
+	RouterSelection
+}
 
-	// RouterService can be used to specify a namespace/name that identifies
-	// peer routers.
-	RouterService ktypes.NamespacedName
+type TemplateRouter struct {
+	WorkingDir         string
+	TemplateFile       string
+	ReloadScript       string
+	DefaultCertificate string
+	RouterService      *ktypes.NamespacedName
+}
 
-	// StatsPort specifies a port at which the router can provide statistics.
-	StatsPort string
+func (o *TemplateRouter) Bind(flag *pflag.FlagSet) {
+	flag.StringVar(&o.WorkingDir, "working-dir", "/var/lib/containers/router", "The working directory for the router plugin")
```

Comment:
Its good that this is configurable but of note here is that the ```images/router/haproxy-base/Dockerfile``` assumes  ```/var/lib/containers/router```. As long as the directory structure is mirrored + certs exist, it should work.